pkg/unidling/api/types.go

```diff
@@ -0,0 +1,20 @@
+package api
+
+import (
+	kextapi "k8s.io/kubernetes/pkg/apis/extensions"
+)
+
+const (
+	IdledAtAnnotation      = "idling.alpha.openshift.io/idled-at"
+	UnidleTargetAnnotation = "idling.alpha.openshift.io/unidle-targets"
+	NeedPodsReason         = "NeedPods"
+)
+
+// RecordedScaleReference is a SubresourceReference to a scale subresource that also
+// has the previous replica count recorded
+type RecordedScaleReference struct {
+	// TODO: update this to a CrossVersionObjectReference once we pull in a Kube version
+	// with an unversioned apis/autoscaling
+	kextapi.SubresourceReference `json:",inline"`
```

Comment:
> but just having a different way to refer to things for this one piece here seems like it would be confusing, and wouldn't help to clarify the distinction between kind and resource.

Spoke with @smarterclayton in person.  I'm not sure if this will make your day better or worse:
 1. Kind is ok as long as you enforce a single RESTMapping based on discovery for your server to locate the correct (and only) resource for a kind.
 1. APIVersion should be split into `Group` and `APIVersion`.  The apiversion is optional, but if it is specified, the group is ignored (this is backwards compatible).  The group is logically required, but empty for legacy groups.  If no apiversion is specified, the code should negotiate for the best available and compatible group version.
This logic is nasty, but we're describing a link, not a serialization format (resource, not kind).  Since group,resource,name represents the same thing across versions, the user wants to describe his intent for the link.  He doesn't want to upgrade his server and have his links break because he was forced to reference a version that eventually got sunset.


I tried and failed to convince him that the semantic for apiVersion was unnecessary for a new API, but if @liggitt wants to try.  Maybe we could even do without apiVersion entirely.